drivers/storage/portworx/util/util.go

```diff
@@ -1902,3 +1905,97 @@ func isVersionSupported(current, target string) bool {
 
 	return currentVersion.Core().GreaterThanOrEqual(targetVersion)
 }
+
+// Get list of storagenodes that are a part of the current cluster that need a node PDB
+func NodesNeedingPDB(cluster *corev1.StorageCluster, sdkConn *grpc.ClientConn, k8sClient client.Client) ([]string, error) {
+	// Get the list of storage nodes
+	nodeClient := api.NewOpenStorageNodeClient(sdkConn)
+	ctx, err := SetupContextWithToken(context.Background(), cluster, k8sClient)
+	if err != nil {
+		return nil, err
+	}
+
+	nodeEnumerateResponse, err := nodeClient.EnumerateWithFilters(
```

Comment:
We are making this call here, in CountStorageNodes and in ClusterSupportsParallelUpgrade. To avoid making 2 portworx calls to get the same data, can we call it once from the pdb component and then pass the output to these functions?